lib/Bindings/Python/OMModule.cpp

```diff
@@ -206,6 +234,33 @@ PythonValue Tuple::getElement(intptr_t i) {
   return omEvaluatorValueToPythonValue(omEvaluatorTupleGetElement(value, i));
 }
 
+PythonValue
+Map::lookup(std::variant<intptr_t, std::string, MlirAttribute> key) {
+  OMEvaluatorValue result;
+  if (auto *i = std::get_if<intptr_t>(&key))
+    result = omEvaluatorMapGetElementByInt(value, *i);
+  else if (auto *str = std::get_if<std::string>(&key))
+    result = omEvaluatorMapGetElementByStr(
+        value, mlirStringRefCreateFromCString(str->c_str()));
+  else
+    result = omEvaluatorMapGetElement(value, std::get<MlirAttribute>(key));
+
+  // Return a fine grained error.
+  if (omEvaluatorValueIsNull(result)) {
+    std::string keyStr;
+    if (auto *i = std::get_if<intptr_t>(&key))
+      keyStr = std::to_string(*i);
+    else if (auto *str = std::get_if<std::string>(&key))
+      keyStr = *str;
+    else
+      keyStr = "attribute";
```

Comment:
Maybe error out, only int and str key type supported ?

Reply:
I'd like to keep it possible to look up via a raw attribute since it's convenient in some cases. This is actually used in python implementation of `Map.__iter__`.